### PR TITLE
Filter ship

### PR DESCRIPTION
**Pull request by NemanjaManot**

### CODE REVIEW

File: src/actions/types.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export const GET_DATA = 'GET_DATA';
 export const GET_DATA_SUCCESS = 'GET_DATA_SUCCESS';
-export const GET_DATA_ERROR = 'GET_DATA_ERROR';
+export const GET_DATA_ERROR = 'GET_DATA_ERROR';
```
Comment (marijarahman) [inline src/actions/types.js]: We agreed to use something more descriptive like SHIP_GET_DATA

File: src/components/FiltersShip/FiltersList.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
+import React, { Component } from 'react';
+import { View, ScrollView, Text } from 'react-native'
+import { Font } from 'expo';
+
+import CheckBox from 'react-native-check-box'
+
+import { FilterGroup } from './FilterGroup';
+import { filterGroup2 } from '../../utils/FilterDropdownGroup';
```
Comment (marijarahman) [inline src/components/FiltersShip/FiltersList.js]: Could you name this differently? FilterGroupHelper maybe

File: src/components/FiltersShip/FiltersList.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            })
+        } else if (groups === 'company') {
+            this.setState({
+                activeFilterGroup3: this.state.activeFilterGroup3 ? false : true
+            })
+        } else if (groups === 'vesselType') {
+            this.setState({
+                activeFilterGroup4: this.state.activeFilterGroup4 ? false : true
+            })
+        }
```
Comment (marijarahman) [inline src/components/FiltersShip/FiltersList.js]: i think switch loop looks cleaner in this case

File: src/components/FiltersShip/FiltersList.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        let arrowDown = <Icon name="keyboard_arrow_down" size={25} color={colors['grayColor']}/>;
+
+        if (groups === 'groups') {
+            return this.state.activeFilterGroup ? arrowUp : arrowDown
+        } else if (groups === 'segment') {
+            return this.state.activeFilterGroup2 ? arrowUp : arrowDown
+        } else if (groups === 'company') {
+            return this.state.activeFilterGroup3 ? arrowUp : arrowDown
+        } else if (groups === 'vesselType') {
+            return this.state.activeFilterGroup4 ? arrowUp : arrowDown
```
Comment (marijarahman) [inline src/components/FiltersShip/FiltersList.js]: also :)

File: src/components/FiltersShip/FiltersList.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        }
+
+    }
+
+    render() {
+
```
Comment (marijarahman) [inline src/components/FiltersShip/FiltersList.js]: Unnecessary space

File: src/components/FiltersShip/FiltersList.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        if (!this.state.fontLoaded) {
```
Comment (marijarahman) [inline src/components/FiltersShip/FiltersList.js]: You could write this in one line

File: src/components/FiltersShip/FooterFiltersShip.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        return (
+            <Footer>
+                <View style={footerIconContainer}>
+                    <View style={footerIconWrapper}>
+                        <Ionicons name="ios-list" size={20} style={styles.footerIcon} />
+                        <Text style={footerIconTitle}>Vessels</Text>
+                    </View>
+
+                    <View style={footerIconWrapper}>
+                        <Ionicons name="ios-funnel" size={20} style={styles.footerIconWarning} />
```
Comment (marijarahman) [inline src/components/FiltersShip/FooterFiltersShip.js]: Desctructure

File: src/Router.js [hunk 8]
```diff
@@ -7,6 +7,7 @@ import CustomNavBar from './shared/CustomNavBar/CustomNavBar';
 import LogIn from './components/LogIn/LogIn';
 import Dashboard from './components/Dashboard/Dashboard'
 import AlertsNotification from './components/AlertsNotifications/AlertNotifications';
+import { FiltersShip } from './components/FiltersShip/FiltersShip';
```
Comment (marijarahman) [inline src/Router.js]: I think you should rename "Ship" to "Vessel"

**Review by marijarahman — CHANGES_REQUESTED** [review]

File: src/Router.js [hunk 9]
```diff
@@ -44,6 +45,17 @@ class RouterComponent extends Component {
                         rightIcon="search"
                         title="ALERTS & NOTIFICATIONS"
                     />
+                    <Scene
+                        key="filters"
+                        sceneStyle={{paddingTop: STATUSBAR_HEIGHT}}
+                        component={FiltersShip}
+                        navBar={CustomNavBar}
+                        icons={true}
+                        leftIcon="keyboard_arrow_down"  // TODO - arrow-left
```
Comment (marko2606) [inline src/Router.js]: I have already added arrow-left on GlobalMap component, when I merge it to dev you can use it too. :)

File: src/components/FiltersVessel/FiltersList.js [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+
+    constructor() {
+        super();
+        this.state = {
+            fontLoaded: false,
+            activeFilterGroup: false,
+            activeFilterGroup2: false,
+            activeFilterGroup3: false,
+            activeFilterGroup4: false
+        }
```
Comment (marko2606) [inline src/components/FiltersVessel/FiltersList.js]: can you name this a little bit different? Maybe something like vesselTypeGroup, companyGroup...

File: src/components/FiltersVessel/FiltersList.js [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+    async componentDidMount() {
+        await Font.loadAsync({
+            'icomoon': require('../../assets/fonts/icomoon.ttf')
+        });
+
+        this.setState({fontLoaded: true});
+    }
+
+    onClickExpandGroup(groups) {
+        switch (groups) {
```
Comment (marko2606) [inline src/components/FiltersVessel/FiltersList.js]: refactor

File: src/components/FiltersVessel/FiltersList.js [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+                    activeFilterGroup4: this.state.activeFilterGroup4 ? false : true
+                });
+                break;
+            default:
+                return groups
+        }
+    }
+
+    iconRender(groups) {
+        let arrowUp = <Icon name="keyboard_arrow_up" size={25} color={colors['grayColor']}/>;
```
Comment (marko2606) [inline src/components/FiltersVessel/FiltersList.js]: refactor

File: src/components/FiltersVessel/FiltersVessel.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { View } from 'react-native';
+
+import FooterFiltersShip from './FiltersVesselFooter';
```
Comment (marko2606) [inline src/components/FiltersVessel/FiltersVessel.js]: different name from imported

File: src/components/FiltersVessel/FiltersVesselFooter.js [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+import { styles } from './FiltersVesselStyle';
+
+class FooterFiltersShip extends Component {
+
+    render() {
+        const { footerIconContainer, footerIconWrapper, footerIconTitle, footerIconTitleWarning } = styles;
+        const iconSize = 20;
+
+        return (
+            <Footer>
```
Comment (marko2606) [inline src/components/FiltersVessel/FiltersVesselFooter.js]: refactor

File: src/Router.js [hunk 15]
```diff
@@ -7,6 +7,7 @@ import CustomNavBar from './shared/CustomNavBar/CustomNavBar';
 import LogIn from './components/LogIn/LogIn';
 import Dashboard from './components/Dashboard/Dashboard'
 import AlertsNotification from './components/AlertsNotifications/AlertNotifications';
+import { FiltersShip } from './components/FiltersVessel/FiltersVessel';
```
Comment (marijarahman) [inline src/Router.js]: Vessel? :)

File: src/actions/types.js [hunk 16]
```diff
@@ -1,3 +1,5 @@
-export const GET_DATA = 'GET_DATA';
-export const GET_DATA_SUCCESS = 'GET_DATA_SUCCESS';
-export const GET_DATA_ERROR = 'GET_DATA_ERROR';
+export const SHIP_GET_DATA = {
```
Comment (marijarahman) [inline src/actions/types.js]: it's supposed to be named VESSEL_ACTION_TYPES

File: src/components/FiltersVessel/FiltersVesselFooter.js (hunk 14 above)
Comment (marijarahman) [reply src/components/FiltersVessel/FiltersVesselFooter.js]: You could create a helper for this (something like menu helper)

**Pull request merged**